massmigration/views.py

```diff
@@ -14,34 +16,48 @@
 def manage_migrations(request):
     """ A page to manage mass migrations. """
     migrations = store.all
-    # Load the migration records in one query to avoid a separate query for each one
-    records_by_key = MigrationRecord.objects.in_bulk()
+    available_db_aliases = get_all_db_aliases()
+
+    # Load the migration in bulk from every db to avoid a separate query for each one
+    migration_records_by_db_alias = {
+        db_alias: MigrationRecord.objects.using(db_alias).in_bulk()
+        for db_alias in available_db_aliases
+    }
+
     for migration in migrations:
-        migration.record = records_by_key.get(migration.key)
+        migration.records_map = OrderedDict()
+
+        for db_alias in available_db_aliases:
+            migration.records_map[db_alias] = {
+                "is_allowed_on_db_alias": db_alias in migration.get_allowed_db_aliases(),
+                "record": migration_records_by_db_alias.get(db_alias).get(migration.key, None)
```

Comment:
```suggestion
                "record": migration_records_by_db_alias[db_alias].get(migration.key, None)
```
Just pedantry/clarity that the first dict will definitely have the `db_alias` key.